### PR TITLE
Replace 'normed' kwarg to hist by 'density' in tests and docs.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

handles https://github.com/matplotlib/matplotlib/pull/10200#discussion_r161638874

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/tests/test_axes.py [hunk 1]
```diff
@@ -2932,7 +2932,7 @@ def test_hist_stacked_normed():
     d2 = np.linspace(0, 10, 50)
     fig = plt.figure()
     ax = fig.add_subplot(111)
-    ax.hist((d1, d2), stacked=True, normed=True)
```
Comment (dstansby) [inline lib/matplotlib/tests/test_axes.py]: I think this should stay as it is, because we should still be testing that `normed` works until it is removed.

Comment (efiring) [reply lib/matplotlib/tests/test_axes.py]: In that case, shouldn't the test have two forms, one for 'normed' and the other for 'density'?  At a more general level, we never have tested everything, and never will--there are too many possible combinations.  We need to concentrate on the most useful tests, putting our resources, and those of the CI services, where they will be most effective in improving mpl.  Testing that a deprecated kwarg still works seems very low priority, doesn't it?  The time of deprecation seems like a perfectly reasonable time to switch a test to the non-deprecated alternative.

Comment (dstansby) [reply lib/matplotlib/tests/test_axes.py]: - There is a test directly below this one for `density`.
- Sure we've never tested everything, but we should *aim* to test everything. This test definitely isn't a CI bottleneck.
- In my opinion, something being deprecated means it is still supported, but there are plans for it to be removed. If it's supported, it should be tested.

I'm not going to block on removing the `normed` test, but I am -2 on it.

Comment (efiring) [reply lib/matplotlib/tests/test_axes.py]: Aiming to test everything doesn't make sense to me, for the reason stated above.

**Comment by anntzer** [comment]
At least the docs should be fixed, I basically don't care about what we do with the tests (but should suppress the warnings if we don't change them).

**Comment by tacaswell** [comment]
I think @dstansby has a good point here.  Can we keep the commit from #10200 for the test file (assuming tests pass, it looks like they are all failing over there) and everything else from this PR?

**Comment by anntzer** [comment]
Feel free to do your own mix and match... or I can rebase after #10200 is merged.

**Comment by QuLogic** [comment]
Oh, didn't know about this. Since #10200 is merged and #10385 took care of the examples, I guess we don't need this PR.

**Pull request closed without merging**